packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol

```diff
@@ -1,226 +1,231 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-import "./legacy_openzeppelin/contracts/access/roles/WhitelistAdminRole.sol";
+import './legacy_openzeppelin/contracts/access/roles/WhitelistAdminRole.sol';
 
 interface ERC20fraction {
-  function decimals() external view returns (uint8);
+    function decimals() external view returns (uint8);
 }
 
 interface AggregatorFraction {
-  function decimals() external view returns (uint8);
-  function latestAnswer() external view returns (int256);
-  function latestTimestamp() external view returns (uint256);
-}
+    function decimals() external view returns (uint8);
+
+    function latestAnswer() external view returns (int256);
 
+    function latestTimestamp() external view returns (uint256);
+}
 
 /**
  * @title ChainlinkConversionPath
  *
  * @notice ChainlinkConversionPath is a contract computing currency conversion rates based on Chainlink aggretators
  */
 contract ChainlinkConversionPath is WhitelistAdminRole {
-  uint constant DECIMALS = 1e18;
-
-  // Mapping of Chainlink aggregators (input currency => output currency => contract address)
-  // input & output currencies are the addresses of the ERC20 contracts OR the sha3("currency code")
-  mapping(address => mapping(address => address)) public allAggregators;
-
-  // declare a new aggregator
-  event AggregatorUpdated(address _input, address _output, address _aggregator);
-
-  /**
-    * @notice Update an aggregator
-    * @param _input address representing the input currency
-    * @param _output address representing the output currency
-    * @param _aggregator address of the aggregator contract
-  */
-  function updateAggregator(address _input, address _output, address _aggregator)
-    external
-    onlyWhitelistAdmin
-  {
-    allAggregators[_input][_output] = _aggregator;
-    emit AggregatorUpdated(_input, _output, _aggregator);
-  }
-
-  /**
-    * @notice Update a list of aggregators
-    * @param _inputs list of addresses representing the input currencies
-    * @param _outputs list of addresses representing the output currencies
-    * @param _aggregators list of addresses of the aggregator contracts
-  */
-  function updateAggregatorsList(
-    address[] calldata _inputs,
-    address[] calldata _outputs,
-    address[] calldata _aggregators
-  )
-    external
-    onlyWhitelistAdmin
-  {
-    require(_inputs.length == _outputs.length, "arrays must have the same length");
-    require(_inputs.length == _aggregators.length, "arrays must have the same length");
-
-    // For every conversions of the path
-    for (uint i; i < _inputs.length; i++) {
-      allAggregators[_inputs[i]][_outputs[i]] = _aggregators[i];
-      emit AggregatorUpdated(_inputs[i], _outputs[i], _aggregators[i]);
+    uint256 constant PRECISION = 1e18;
+    uint256 constant NATIVE_TOKEN_DECIMALS = 18;
```

Comment:
Can this be different for another chain? I can't find the spec for this.

Reply:
It's not completely impossible but unlikely. If it happens we can anticipate and adapt the contract before we deploy it (either with a hardcoded value or add a constructor parameter)